### PR TITLE
Nim implementation

### PR DESCRIPTION
**Pull request by marghidanu**
## Description

Nim implementation

## Contributing requirements

* [x] I read the contribution guidelines in CONTRIBUTING.md.
* [x] I placed my solution in the correct category folder.
* [x] I added a README.md or Dockerfile.
* [x] I selected `drag-race` as the target branch.
* [x] All code herein is licensed compatible with BSD-3.

The numbers look pretty good on my M1 MacBook Air, check this out:

```
Passes: 8979, Time: 5.000272035598755, Avg: 0.0005568851804876662, Limit: 1000000, Count1: 78498, Count2: 78498, Valid: true
Passes: 9424, Time: 5.000049114227295, Avg: 0.0005305654832584142, Limit: 1000000, Count1: 78498, Count2: 78498, Valid: true
Passes: 9040, Time: 5.000542879104614, Avg: 0.0005531573981310413, Limit: 1000000, Count1: 78498, Count2: 78498, Valid: true
Passes: 9057, Time: 5.000289916992188, Avg: 0.000552091191011614, Limit: 1000000, Count1: 78498, Count2: 78498, Valid: true
Passes: 8979, Time: 5.000046014785767, Avg: 0.0005568600083289639, Limit: 1000000, Count1: 78498, Count2: 78498, Valid: true
Passes: 9309, Time: 5.000034093856812, Avg: 0.0005371182827217544, Limit: 1000000, Count1: 78498, Count2: 78498, Valid: true
Passes: 9126, Time: 5.000533103942871, Avg: 0.0005479435792179346, Limit: 1000000, Count1: 78498, Count2: 78498, Valid: true
Passes: 9135, Time: 5.000319004058838, Avg: 0.0005473802960108197, Limit: 1000000, Count1: 78498, Count2: 78498, Valid: true
Passes: 8687, Time: 5.000458955764771, Avg: 0.0005756255273126246, Limit: 1000000, Count1: 78498, Count2: 78498, Valid: true
Passes: 8541, Time: 5.000423908233643, Avg: 0.0005854611764703948, Limit: 1000000, Count1: 78498, Count2: 78498, Valid: true
```

### CODE REVIEW

File: PrimeNim/solution_1/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+Passes: 9424, Time: 5.000049114227295, Avg: 0.0005305654832584142, Limit: 1000000, Count1: 78498, Count2: 78498, Valid: true
+Passes: 9040, Time: 5.000542879104614, Avg: 0.0005531573981310413, Limit: 1000000, Count1: 78498, Count2: 78498, Valid: true
+Passes: 9057, Time: 5.000289916992188, Avg: 0.000552091191011614, Limit: 1000000, Count1: 78498, Count2: 78498, Valid: true
+Passes: 8979, Time: 5.000046014785767, Avg: 0.0005568600083289639, Limit: 1000000, Count1: 78498, Count2: 78498, Valid: true
+Passes: 9309, Time: 5.000034093856812, Avg: 0.0005371182827217544, Limit: 1000000, Count1: 78498, Count2: 78498, Valid: true
+Passes: 9126, Time: 5.000533103942871, Avg: 0.0005479435792179346, Limit: 1000000, Count1: 78498, Count2: 78498, Valid: true
+Passes: 9135, Time: 5.000319004058838, Avg: 0.0005473802960108197, Limit: 1000000, Count1: 78498, Count2: 78498, Valid: true
+Passes: 8687, Time: 5.000458955764771, Avg: 0.0005756255273126246, Limit: 1000000, Count1: 78498, Count2: 78498, Valid: true
+Passes: 8541, Time: 5.000423908233643, Avg: 0.0005854611764703948, Limit: 1000000, Count1: 78498, Count2: 78498, Valid: true
```
Comment (rbergen) [inline PrimeNim/solution_1/README.md:41]: That's astonishingly quick, indeed!

File: PrimeNim/solution_1/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+```
+
+## Notes
+
+I somehow feel that this is the type of language I would have a love-hate relationship.
```
Comment (rbergen) [inline PrimeNim/solution_1/README.md:46]: I would be interested to know why that is. :)

**Review by rbergen — APPROVED** [review]
LGTM

File: PrimeNim/solution_1/README.md (hunk 2 above)
Comment (marghidanu) [reply PrimeNim/solution_1/README.md:46]: For example, some things like hashes are not built-in in the language, you have to use an import to get tables. Also, defining classes feels weird.

**Pull request merged**